### DISCUSSION TITLE
[bug]: build lnd 0.18.4 failure: package slices is not in GOROOT

### DISCUSSION THREAD

**Discussion by alevchuk** [lightningnetwork/lnd#9327] [Troubleshooting]
### Background

I'm following [Building a development version from source](https://github.com/lightningnetwork/lnd/blob/master/docs/INSTALL.md), yet I got two errors. I was able to fix the first one, but got stuck on the second.

First error:
```
$ (cd ~/src/lnd && make clean && make install)
/home/lightning/src/lnd/go.mod:212: invalid go version '1.21.4': must match format 1.23
```
I overcame it by editing go.mod and replacing '1.21.4' with '1.21'

Second error:
```
$ (cd ~/src/lnd && make clean && make install)
contractcourt/chain_watcher.go:6:2: package slices is not in GOROOT (/home/lightning/src/go/src/slices)
```

* it happens right after "go: downloading github.com/remyoudompheng/bigfft v0.0.0-20230129092748-24d4a6f8daec"

### Your environment

* version of `lnd`: d72a3aaf261e278fa4aad5be4453df2f74ab50ee  tag: v0.18.3-beta
* version of `go`: 619b8fd7d2c94af12933f409e962b99aa9263555 tag: go1.19.13
* which operating system Debian 11.11: `Linux 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux`
* any other relevant environment details

My paths are setup like this:
```
export GOROOT_BOOTSTRAP="$(ls -1d /usr/lib/go-* | tail -n1)"
export GOROOT=~/src/go
export GOPATH=~/gocode
export PATH=$GOROOT/bin:$GOPATH/bin:$PATH
```

### Steps to reproduce

1. build go1.19.13
2. build LND v0.18.3-beta

**Comment by alevchuk** [comment]
realized i had go 1.18 but "lnd is written in Go, with a minimum version of 1.19"

**Comment by alevchuk** [comment]
Actually I'm getting the same on go1.19.13

**Reply by guggero** [reply]
The [installation documentation](https://github.com/lightningnetwork/lnd/blob/master/docs/INSTALL.md) is out of date. The minimum version you need to compile `lnd` is 1.22 as per the `go.mod` file. I'll update the docs.

**Comment by ziggie1984** [comment]
> contractcourt/chain_watcher.go:6:2: package slices is not in GOROOT (/home/lightning/src/go/src/slices)

seems like your golang installation is not correct.

try to compile a simple golang program and see whether this works.

GOROOT needs to point to the directory where the golang implementation lies. Normally this is set automatically and is normally `/usr/local/go` for linux systems.
